python_modules/libraries/dagster-gcp/dagster_gcp/solids.py

```diff
@@ -0,0 +1,302 @@
+            expiration_ms=time_partitioning['expiration_ms'],
+            require_partition_filter=time_partitioning['require_partition_filter'],
+        )
+    return QueryJobConfig(**kwargs) if kwargs else None
+
+
+def _extract_load_job_config(cfg):
+    destination_encryption_configuration = cfg.get('destination_encryption_configuration')
+    time_partitioning = cfg.get('time_partitioning')
+
+    fields = [
+        'allow_jagged_rows',
+        'allow_quoted_newlines',
+        'autodetect',
+        'clustering_fields',
+        'create_disposition',
+        'destination_encryption_configuration',
+        'destination_table_description',
+        'destination_table_friendly_name',
+        'encoding',
+        'field_delimiter',
+        'ignore_unknown_values',
+        'max_bad_records',
+        'null_marker',
+        'quote_character',
+        'schema_update_options',
+        'skip_leading_rows',
+        'source_format',
+        'time_partitioning',
+        'use_avro_logical_types',
+        'write_disposition',
+    ]
+    kwargs = {k: cfg.get(k) for k in fields if cfg.get(k) is not None}
+
+    if destination_encryption_configuration is not None:
+        kwargs['destination_encryption_configuration'] = EncryptionConfiguration(
+            kms_key_name=destination_encryption_configuration
+        )
+
+    if time_partitioning is not None:
+        kwargs['time_partitioning'] = TimePartitioning(
+            field=time_partitioning['field'],
+            expiration_ms=time_partitioning['expiration_ms'],
+            require_partition_filter=time_partitioning['require_partition_filter'],
+        )
+    return LoadJobConfig(**kwargs) if kwargs else None
+
+
+class BigQuerySolidDefinition(SolidDefinition):
+    """
+    Executes BigQuery SQL queries.
+    """
+
+    def __init__(self, name, sql_queries, description=None):
+        name = check.str_param(name, 'name')
+        sql_queries = check.list_param(sql_queries, 'sql queries', of_type=str)
+        description = check.opt_str_param(description, 'description', 'BigQuery query')
+
+        def _transform_fn(context, _):
+            (project, location) = [context.solid_config.get(k) for k in ('project', 'location')]
+            query_job_config = context.solid_config.get('query_job_config', {})
+
+            client = bigquery.Client(project=project, location=location)
+
+            # Retrieve results as pandas DataFrames
+            results = []
+            for sql_query in sql_queries:
+                # We need to construct a new QueryJobConfig for each query.
+                # See: https://bit.ly/2VjD6sl
+                cfg = _extract_query_job_config(query_job_config)
+                context.log.info(
+                    'executing query %s with config: %s'
+                    % (sql_query, cfg.to_api_repr() if cfg else '(no config provided)')
+                )
+
+                result = client.query(sql_query, job_config=cfg).to_dataframe()
```

Comment:
this definitely feels like it should be more pluggable/composable. all the other logic here is quite generic. I assume there are other to_XXX in the API

Reply:
This is pretty much it as far as queries go. The only alternative is to use `.result()` to get back `google.cloud.bigquery.table.Row` instances, which aren't particularly useful:

```
from google.cloud import bigquery
client = bigquery.Client()

sql = """
    SELECT name, SUM(number) as count
    FROM `bigquery-public-data.usa_names.usa_1910_current`
    GROUP BY name
    ORDER BY count DESC
    LIMIT 10
"""
print(list(client.query(sql).result()))


[Row(('James', 5001762), {'name': 0, 'count': 1}),
 Row(('John', 4875934), {'name': 0, 'count': 1}),
 Row(('Robert', 4743843), {'name': 0, 'count': 1}),
```